packages/rrweb/src/plugins/network/record/index.ts

```diff
@@ -0,0 +1,491 @@
+      const networkRequest: Partial<NetworkRequest> = {};
+      let after: number | undefined;
+      let before: number | undefined;
+      try {
+        const requestHeaders: Headers = {};
+        req.headers.forEach((value, header) => {
+          requestHeaders[header] = value;
+        });
+        if (recordRequestHeaders) {
+          networkRequest.requestHeaders = requestHeaders;
+        }
+        if (shouldRecordBody('request', options.recordBody, requestHeaders)) {
+          if (req.body === undefined || req.body === null) {
+            networkRequest.requestBody = null;
+          } else {
+            networkRequest.requestBody = req.body;
+          }
+        }
+        after = win.performance.now();
+        res = await originalFetch(req);
+        before = win.performance.now();
+        const responseHeaders: Headers = {};
+        res.headers.forEach((value, header) => {
+          responseHeaders[header] = value;
+        });
+        if (recordResponseHeaders) {
+          networkRequest.responseHeaders = responseHeaders;
+        }
+        if (shouldRecordBody('response', options.recordBody, responseHeaders)) {
+          let body: string | undefined;
+          try {
+            body = await res.clone().text();
+          } catch {
+            //
+          }
+          if (res.body === undefined || res.body === null) {
+            networkRequest.responseBody = null;
+          } else {
+            networkRequest.responseBody = body;
+          }
+        }
+        return res;
+      } finally {
+        getRequestPerformanceEntry(win, 'fetch', req.url, after, before)
+          .then((entry) => {
+            const request: NetworkRequest = {
+              url: entry.name,
+              method: req.method,
+              initiatorType: entry.initiatorType as InitiatorType,
+              status: res?.status,
+              startTime: Math.round(entry.startTime),
+              endTime: Math.round(entry.responseEnd),
+              requestHeaders: networkRequest.requestHeaders,
+              requestBody: networkRequest.requestBody,
+              responseHeaders: networkRequest.responseHeaders,
+              responseBody: networkRequest.responseBody,
+            };
+            cb({ requests: [request] });
+          })
+          .catch(() => {
+            //
+          });
+      }
+    };
+  });
+  return () => {
+    restorePatch();
+  };
+}
+
+function initNetworkObserver(
+  callback: networkCallback,
+  win: IWindow, // top window or in an iframe
+  options: NetworkRecordOptions,
+): listenerHandler {
+  if (!('performance' in win)) {
+    return () => {
+      //
+    };
+  }
+  const networkOptions = (options
+    ? Object.assign({}, defaultNetworkOptions, options)
+    : defaultNetworkOptions) as Required<NetworkRecordOptions>;
+
+  const cb: networkCallback = (data) => {
+    const requests = data.requests.filter(
+      (request) => !networkOptions.ignoreRequestFn(request),
+    );
```

Comment:
It would be great if this was both filtering and cleansing.

The most typical case for privacy controls is:
"I want to filter out requests to this place"
"I wan't to ensure certain headers aren't included"
"I want to filter out PII from the response bodies".

This would all doable with a slight modification of the function:
```suggestion
    const requests = data.requests.reduce(
      (list, request) => {
        const maskedRequest = networkOptions.maskRequestFn(request)
        
        return maskedRequest ? [...list, maskedRequest] : list
      }, []
    );
```

Which then allows the function to be something like

```ts
maskRequestFn: (request: NetworkRequest): NetworkRequest | undefined  => {
  if (request.url.contains("rrweb-collector-api.com") {
    return
 }
 delete request.requestHeaders["Authorization"]
 request.responseBody = maskTextFn(request.responseBody)
 return request
}

```

Reply:
In theory this could be added in a follow up PR as well. Happy to do that if that is the decision.

Reply:
Fixed here: https://github.com/rrweb-io/rrweb/pull/1689